File: app/src/main/java/com/georgcantor/wallpaperapp/network/ApiService.java
package com.georgcantor.wallpaperapp.network;

import com.georgcantor.wallpaperapp.BuildConfig;
import com.georgcantor.wallpaperapp.model.Pic;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface ApiService {

    @GET("?key=" + BuildConfig.API_KEY + "&q=mercedes-benz")
    Call<Pic> getMercedesPic(@Query("page") int index);

    @GET("?key=" + BuildConfig.API_KEY)
    Call<Pic> getCatPic(@Query("category") String category,
                        @Query("page") int index);

    @GET("?key=" + BuildConfig.API_KEY + "&q=bmw")
    Call<Pic> getBmwPic(@Query("page") int index);

    @GET("?key=" + BuildConfig.API_KEY + "&q=porsche")
    Call<Pic> getPorschePic(@Query("page") int index);

    @GET("?key=" + BuildConfig.API_KEY + "&q=audi")
    Call<Pic> getAudiPic(@Query("page") int index);

    @GET("?key=" + BuildConfig.API_KEY + "&q=bugatti")
    Call<Pic> getBugattiPic(@Query("page") int index);

    @GET("?key=" + BuildConfig.API_KEY + "&q=ferrari")
    Call<Pic> getFerrariPic(@Query("page") int index);

    @GET("?key=" + BuildConfig.API_KEY + "&q=lamborghini")
    Call<Pic> getLamboPic(@Query("page") int index);
}

File: app/src/main/java/com/georgcantor/wallpaperapp/model/ApiService.kt
package com.georgcantor.wallpaperapp.model

import com.georgcantor.wallpaperapp.BuildConfig.PIXABAY_KEY
import com.georgcantor.wallpaperapp.model.response.pixabay.Pic
import com.georgcantor.wallpaperapp.model.response.unsplash.UnsplashResponse
import com.georgcantor.wallpaperapp.model.response.videos.VideoResponse
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Query
import retrofit2.http.Url

interface ApiService {

    @GET("?key=$PIXABAY_KEY")
    suspend fun getPixabayPictures(
        @Query("q") query: String,
        @Query("page") index: Int
    ): Response<Pic>

    @GET
    suspend fun getUnsplashPictures(
        @Url url: String,
        @Query("query") query: String,
        @Query("page") page: Int
    ): Response<UnsplashResponse>

    @GET
    suspend fun getVideos(
        @Url url: String,
        @Query("playlistId") playlistId: String
    ): Response<VideoResponse>
}